lib/rules/no-floating-promise.js

```diff
@@ -105,62 +159,14 @@ module.exports = {
 
     create(context) {
         return {
-
-            /*
-             * Note: CallExpression does not match on "await foo()" as this is an AwaitExpression
-             * so this condition already filters out non-floating promises
-             */
+            "VariableDeclarator > CallExpression"(node) {
```

Comment:
Other than refactoring, I think this `VariableDeclarator > CallExpression` is the only thing this PR adds